http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -175,40 +180,47 @@ class ClusterMonitorActor(
   /**
    * Handles a dataproc cluster which is not ready yet. We don't take any action, just
    * schedule another monitor pass.
-   * @param status the ClusterStatus from Google
+   * @param cluster the cluster being monitored
+   * @param googleStatus the ClusterStatus from Google
+   * @param googleInstances the cluster's instances in Google
    * @return ScheduleMonitorPass
    */
-  private def handleNotReadyCluster(status: ClusterStatus,
-                                    instances: Set[Instance],
+  private def handleNotReadyCluster(cluster: Cluster,
+                                    googleStatus: ClusterStatus,
+                                    googleInstances: Set[Instance],
                                     msg: Option[String]): Future[ClusterMonitorMessage] = {
-    val currTimeElapsed: Long = (System.currentTimeMillis() - startTime)
-
-    if (monitorConfig.monitorStatusTimeouts.keySet.contains(status) &&
-        currTimeElapsed > monitorConfig.monitorStatusTimeouts(status).toMillis) {
-
-      logger.info(
-        s"Detected that ${cluster.projectNameString} has been stuck in status $status too long. Failing it. Current timeout config: ${monitorConfig.monitorStatusTimeouts.toString}"
-      )
+    val currTimeElapsed: FiniteDuration = (System.currentTimeMillis() - startTime).millis
+    monitorConfig.monitorStatusTimeouts.get(cluster.status) match {
+      case Some(timeLimit) if currTimeElapsed > timeLimit =>
+        logger.info(s"Detected that ${cluster.projectNameString} has been stuck in status ${cluster.status} too long.")
+
+        // Take care not to Error out a cluster if it timed out in Starting status
```

Comment:
Because `Error` is an unrecoverable status. We don't want a user to start their cluster; have it time out; then have Leo error out (and delete) their cluster, potentially losing data.

This code transitions the cluster back to `Stopped`, so the user can try again or create a ticket -- either way, we won't delete their data.

Reply:
got it